R/assign.R

```diff
@@ -0,0 +1,473 @@
+    # each row in the dataset
+    dat$ghost = NA
+    dat$trackID = NA
+    dat$rametArea = NA
+  inv <- sort(sampleInv) ## integer vector of quadrat sampling years in
+  # sequential order
+  ## make sure that the 'assignOut' data.frame that contains the output is empty
+  rm(assignOut)
+  ## user-defined arguments
+  dorm <- 0 ## dormancy allowed by the function
+  buff <- .05 ## buffer of movement allowed from year to year, in meters
+  buffGenet <- 0.001 ## buffer between polygons (/2) that is the maximum allowed for
+  # them to be considered genets
+  # overlap <- .50 ## the percentage of overlap (in decimal form) between focal
+  # indiv. and next year indiv. that will be required to consider them both
+  # the same individual (not 100% sure on this one yet...)
+  clonal <- 0 ## binary option that indicates whether this species is allowed to
+  # break into multiple polygons for one individual
+
+  #conalBuff <- .01 ## the buffer of overlap that indicates polygons are genets
+  ## of the same individual (if genets are allowed), in meters ##probably don't
+  ##need...
+
+  ## work -------------------------------------------------------------------
+  datFirst <- dat[dat$Year==inv[1],] ## get the data just for the first year of
+  # sampling
+  if (clonal==1) { ## if this species is 'clonal' (clonal argument == 1), use
+    # Dave's groupByGenet function to give a singleTrackID to a group of
+    # polygons if they are close enough to one another (user-defined)
+    datFirst$genetID <- groupByGenet(datFirst, buffGenet) # put these temporary
+    # genet ID's in a temporary 'genetID' column, which will tell us to give
+    # these genets the same track ID
+    ## aggregate size by genetID (total size for each ramet)
+    datFirst <- st_join(datFirst, (datFirst %>%
+      group_by(genetID) %>%
+      summarize(rametAreaTemp = sum(Area))), by = "genetID") %>%
+      select(-c(genetID.x, rametArea)) %>%
+      rename("genetID" = "genetID.y", "rametArea" = "rametAreaTemp")
+  }
+  if (clonal == 0) { ## if genets are not allowed (clonal == 0), then each
+    # individual polygon gets a unique value in the 'genetID' column
+    datFirst$genetID <- 1:nrow(datFirst)
+    datFirst$rametArea <- datFirst$Area
+  }
+
+  ## assign a unique trackID to every unique genetID in this first-year dataset
+  IDs <- data.frame( "genetID" = sort(unique(datFirst$genetID)), ## get a vector
+                     # of all of the unique genetIDs
+            "trackID" = paste0(unique(dat$sp_code_6), ## get the unique 6-letter
+                               # species code
+                               "_",unique(datFirst$Year), ## get the unique year
+                  "_",c(1:length(unique(datFirst$genetID))))) ## get a vector of
+  # unique numbers that is the same length as the genetIDs in this quad/year
+
+  datFirst<- merge(datFirst[,names(datFirst) != "trackID"], IDs, by = "genetID")
```

Comment:
Since this only appears to add on one column it could be done with a match statement instead. Food for thought.

Reply:
not quite sure how to do this?